core/src/main/java/org/lflang/TargetProperty.java

```diff
@@ -465,6 +465,13 @@ public enum TargetProperty {
           config.platformOptions = new PlatformOptions();
           config.platformOptions.platform =
               (Platform) UnionType.PLATFORM_UNION.forName(ASTUtils.elementToSingleString(value));
+          if (config.platformOptions.platform == null) {
+            String s =
+                "Unidentified Platform Type, LF supports the following platform types: "
+                    + Arrays.asList(Platform.values()).toString();
+            err.at(value).error(s);
+            throw new AssertionError(s);
```

Comment:
This should be checked in the validator.

Reply:
This pattern was copied from other target properties. I see why, because it's currently difficult to do it the "correct" way, and it introduces a bunch of code duplication. I think it might be worth to try and improve this situation... I'm currently trying to chip away at that a little bit...

Reply:
I will push my refactoring of this to another branch. Let's merge this PR first.